pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter.go

```diff
@@ -156,7 +156,7 @@ func ValidateVirtualMachineInstanceSpec(field *k8sfield.Path, spec *v1.VirtualMa
 		return appendNewStatusCauseForMaxNumberOfVolumesExceeded(field, causes)
 	}
 
-	causes = append(causes, validateHostNameNotConformingToDNSLabelRules(field, spec)...)
+	causes = append(causes, validateHostNameNotConformingToDNSLabelRules(field, spec.Hostname)...)
```

Comment:
With the current implementation, the resulting cause will be wrong.
The `field.Child("hostname")` in `appendNewStatusCauseForHostNameNotConformingToDNSLabelRules` will fail.
As I already suggested,  this change looks good to me, it is the function that should be adjusted.